### PR TITLE
docs: describe interaction between editorconfig and ftplugin

### LINKED ISSUE #28484: Docs: describe priority between editorconfig and autocmd like FileType
### Problem

`:h editorconfig` does not explain when and how the settings are applied, jus that it is automatic (in nvim 0.9.4).

It seems like it applies FileType autocmd first and then editorconfig after that - so that editorconfig has precedence. To override the editorconfig one can explicitly do `:set ft=python` for example.

### Expected behavior

The behavior should be described in the help page.

### PR DESCRIPTION
**Pull request by dundargoc**
Closes https://github.com/neovim/neovim/issues/28484.

Co-authored-by: Zane Dufour <zane@znd4.me>

### CODE REVIEW

File: runtime/lua/editorconfig.lua [hunk 1]
```diff
@@ -1,10 +1,9 @@
 --- @brief
---- Nvim supports EditorConfig. When a file is opened, Nvim searches all parent
---- directories of that file for ".editorconfig" files, parses them, and applies
---- any properties that match the opened file. Think of it like 'modeline' for an
---- entire (recursive) directory. For more information see
---- https://editorconfig.org/.
----
+--- Nvim supports EditorConfig. When a file is opened, after running |ftplugin|s,
+--- Nvim searches all parent directories of that file for ".editorconfig" files,
```
Comment (gpanders) [inline runtime/lua/editorconfig.lua:3]: Should mention `FileType` autocommands too.

```suggestion
--- Nvim supports EditorConfig. When a file is opened, after running |ftplugin|s and |FileType| autocommands,
--- Nvim searches all parent directories of that file for ".editorconfig" files,
```

File: runtime/lua/editorconfig.lua [hunk 2]
```diff
@@ -1,10 +1,9 @@
 --- @brief
---- Nvim supports EditorConfig. When a file is opened, Nvim searches all parent
---- directories of that file for ".editorconfig" files, parses them, and applies
---- any properties that match the opened file. Think of it like 'modeline' for an
---- entire (recursive) directory. For more information see
---- https://editorconfig.org/.
----
+--- Nvim supports EditorConfig. When a file is opened, after running |ftplugin|s
```
Comment (dundargoc) [inline runtime/lua/editorconfig.lua:2]: Is this too wordy?

```suggestion
--- Nvim supports EditorConfig. When a file is opened, after running |ftplugin|s (including the ones in `after/`)
```

Comment (gpanders) [reply runtime/lua/editorconfig.lua:2]: I don't think we need this, `after/ftplugin` is still part of "ftplugins".

**Review by gpanders — APPROVED** [review]

**Comment by dundargoc** [comment]
Thank you for your contribution! We collect all small documentation fixes into a single pull request and merge it once it gets big enough: https://github.com/neovim/neovim/pull/28609

**Comment by clason** [comment]
image
